influxql/sanitize.go

```diff
@@ -0,0 +1,43 @@
+package influxql
+
+import (
+	"bytes"
+	"regexp"
+)
+
+// Sanitize attempts to sanitize passwords out of a raw query.
+// It looks for patterns that may be related to the SET PASSWORD and CREATE USER
+// statements and will redact the password that should be there. It will attempt
+// to redact information from common invalid queries too, but it's not guaranteed
+// to succeed on improper queries.
+//
+// This function works on the raw query and attempts to retain the original input
+// as much as possible.
+func Sanitize(query string) string {
+	sanitizeSetPassword := regexp.MustCompile(`(?i)password\s+for[^=]*=\s+(["']?[^\s"]+["']?)`)
```

Comment:
It should be. Let me move it. This was just easier while I was writing the code.